### PR TITLE
Fix THENINJARPG-2F8: Prevent infinite loop in Countdown component

### LINKED ISSUE #1041: [THENINJARPG-2F8] Maximum update depth exceeded in Countdown component
**Sentry Issue:** [THENINJARPG-2F8](https://studie-tech-aps.sentry.io/issues/THENINJARPG-2F8)

The Countdown component in `app/src/layout/Countdown.tsx` causes an infinite loop error when the countdown finishes and its `onFinish` callback triggers data refetch.

**Error Message:**
```
Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate.
```

**Affected Page:** /manual/polls

**Root Cause:**
When a countdown finishes and calls `onFinish`, the callback typically invalidates/refetches data, causing a parent component re-render. If the re-render causes `targetDate` to change (even slightly, like when using `|| new Date()` as fallback), the `hasCalledOnFinishRef` flag gets reset, triggering another `onFinish` call and creating an infinite loop.

**Sentry Link:** https://studie-tech-aps.sentry.io/issues/91302439/

### PR DESCRIPTION
**Pull request by MathiasGruber**
## Summary
Add conditional check to only reset hasCalledOnFinishRef when the new countdown target is in the future, preventing infinite loops when targetDate changes after onFinish triggers data invalidation.

## Why
**Sentry Issue**: [THENINJARPG-2F8](https://studie-tech-aps.sentry.io/issues/91302439/)

**Error observed**: Maximum update depth exceeded - React infinite loop in Countdown component

**Frequency**: 1 event (new issue)

**Key insight from Sentry**: Stack trace pointed directly to `Countdown.tsx` at the `updateCountdown` function, with mechanism `auto.browser.browserapierrors.setInterval` confirming interval-related error. The error occurred during a sign-out action on /manual/polls page.

**Root cause**: When a countdown finishes and calls `onFinish`, it triggers data invalidation which can cause re-renders with slightly different `targetDate` values. The code was unconditionally resetting `hasCalledOnFinishRef` to false whenever `targetTime` changed, causing `onFinish` to be called again since the countdown was still done. This created an infinite loop.

**Fix**: Only reset the `hasCalledOnFinishRef` flag when the new countdown target is actually in the future (indicating a genuinely new countdown), not when it's in the past (indicating just a slight variation of an already-completed countdown).

**Sentry Issue:** [THENINJARPG-2F8](https://studie-tech-aps.sentry.io/issues/THENINJARPG-2F8)

## Test plan
- Verified locally
- Code review passed

Closes #1041

---

> [!NOTE]
> **Low Risk**
> Small, localized logic change in a UI component; main risk is edge-case behavior around resetting completed countdowns.
> 
> **Overview**
> Prevents `Countdown` from repeatedly firing `onFinish` after completion by changing when the internal “finished” flag resets.
> 
> The reset of `hasCalledOnFinishRef` was moved into the `useEffect` and is now conditional: it only resets when `targetTime` changes *and* the new target is still in the future, avoiding re-render/update loops when completed countdowns receive slightly different past `targetDate` values.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 2fc78eafec2c32c4823dfd260dd6e2f84f1cd1cf. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved countdown behavior when the target time changes. The countdown no longer resets its completion state unnecessarily when the target time is updated after the countdown has already finished. This prevents potential infinite loop scenarios and makes the component more robust when managing dynamic target times.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: app/src/layout/Countdown.tsx [hunk 1]
```diff
@@ -24,10 +24,14 @@ const Countdown: React.FC<CountdownProps> = (props) => {
...
-  // Reset finish flag when target changes
+  // Reset finish flag when target changes - but only if the new countdown is in the future
   if (prevTargetTimeRef.current !== targetTime) {
     prevTargetTimeRef.current = targetTime;
-    hasCalledOnFinishRef.current = false;
+    // Only reset onFinish flag if the new countdown is actually in the future
+    // This prevents infinite loops when targetDate changes after countdown completes
+    if (targetTime > Date.now()) {
+      hasCalledOnFinishRef.current = false;
+    }
```
[the remark this answers is not in the document]
Comment (MathiasGruber) [reply app/src/layout/Countdown.tsx]: Fixed! The conditional ref reset logic has been moved from the render path into the existing useEffect that depends on targetTime. All hooks are now unconditionally at the top level before any conditional logic.

— Applied by Claude on behalf of nano.mathias@gmail.com
